sql: add a note about `FULLTEXT` (#17876)
pingcap/docs: sql-statements/sql-statement-add-index.md

## MySQL compatibility

* TiDB accepts index types such as `HASH`, `BTREE` and `RTREE` in syntax but ignores them for compatibility reasons.
* `SPATIAL` indexes are not supported.

> * `FULLTEXT` is only for syntax compatibility with MySQL and is not supported.

Author: ```suggestion
```

> * TiDB supports parsing the `FULLTEXT` syntax but does not support using the `FULLTEXT` indexes.

Reviewer: I think lines 90 and 91 are saying the same thing now

Author: Ok deleted one

## MySQL compatibility

> * `HASH` and `SPATIAL` indexes are not supported.

Reviewer: ```suggestion
* Index types like `HASH`, `BTREE` and `RTREE` are accepted in syntax but ignore for compatibility reasons.
* `SPATIAL` indexes are not supported.
```

> * `FULLTEXT` is only for syntax compatibility with MySQL and is not supported.

Reviewer: ```suggestion
* `FULLTEXT` is only for syntax compatibility with MySQL and is not supported.
* TiDB supports parsing the `FULLTEXT` syntax but does not support using the `FULLTEXT` indexes.
```

## MySQL compatibility

> * TiDB accepts index types such as `HASH`, `BTREE` and `RTREE` in syntax but ignores them for compatibility reasons.
> * `SPATIAL` indexes are not supported.
> * TiDB supports parsing the `FULLTEXT` syntax but does not support using the `FULLTEXT` indexes.

Reviewer 2: ```suggestion
* TiDB accepts index types such as `HASH`, `BTREE` and `RTREE` in syntax for compatibility with MySQL, but ignores them.
* `SPATIAL` indexes are not supported.
* TiDB supports parsing the `FULLTEXT` syntax but does not support using the `FULLTEXT` indexes.
```
